### PR TITLE
fix: fp with user-agent containing ; pg (932239 PL2)

### LINKED ISSUE #3725: 932239 blocking Mozilla user-agents | Matched Data: ; PG found
### Description
We have a lot of absolutely legal requests from real users with User-Agent like this:
`Mozilla/5.0 (Linux; Android 14; PGT-N19 Build/HONORPGT-N49; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/124.0.6367.180 Mobile Safari/537.36`

These requests are blocked by 932239 with reason:
`Matched Data: ; PG found within REQUEST_HEADERS:user-agent: Mozilla/5.0 (Linux; Android 14; PGT-N19 Build/HONORPGT-N49; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/124.0.6367.180 Mobile Safari/537.36`

### How to reproduce the misbehavior (-> curl call)
```
curl -X GET host.with.pl2.enabled -kL \
-H "User-Agent: Mozilla/5.0 (Linux; Android 14; PGT-N19 Build/HONORPGT-N49; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/124.0.6367.180 Mobile Safari/537.36" \
-H "Host: host.with.pl2.enabled"
```

### Your Environment

* CRS version (e.g., v3.3.4):4.3.0
* Paranoia level setting (e.g. PL1) :PL2
* ModSecurity version (e.g., 2.9.6): ModSecurity v3.0.12 (Linux)
* Web Server and version or cloud provider / CDN (e.g., Apache httpd 2.4.54): ingress-nginx controller v1.10.0
* Operating System and version: n/a

### Confirmation

[ ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

### PR DESCRIPTION
**Pull request by franbuehler**
This PR resolves #3725 by adding a `@` to `pg` to the file [unix-shell-upto3.ra](https://github.com/coreruleset/coreruleset/blob/main/regex-assembly/include/unix-shell-upto3.ra#L150). It also adds a test.

### CODE REVIEW

File: tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932239.yaml [hunk 1]
```diff
@@ -903,3 +903,19 @@ tests:
...
+          input:
+            dest_addr: 127.0.0.1
+            headers:
+              Accept: "text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5"
+              Host: localhost
+              User-Agent: "Mozilla/5.0 (Linux; Android 14; PGT-N19 Build/HONORPGT-N49; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/124.0.6367.180 Mobile Safari/537.36"
+            method: POST
+            port: 80
+            uri: "/post"
+            version: HTTP/1.0
```
Comment (azurit) [inline tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932239.yaml:919]: Let's set this to HTTP/1.1.

**Comment by azurit** [comment]
What exactly will this PR cause? Will it exclude `pg`, `pg@` and `pg~` from regex assembling?

File: tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932239.yaml (hunk 1 above)
Comment (franbuehler) [reply tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932239.yaml:919]: Thank you. I'll change it.

**Comment by franbuehler** [comment]
> What exactly will this PR cause? Will it exclude `pg`, `pg@` and `pg~` from regex assembling?

Yes, exactly.

**Comment by azurit** [comment]
That doesn't seems fully correct to me, as we are having FPs with `PGT-` and not with, for example, `pg ` (with a space at the end).

**Comment by franbuehler** [comment]
Yes, true. So, to match `pgt` but not `pg`, I'll add `~` but not `@` to the exclusion file, right?

**Comment by franbuehler** [comment]
I don't know why test 932300 is failing. I had to run `crs-toolchain regex update --all` and used the latest crs-toolchain.

**Review by fzipi — APPROVED** [review]
LGTM

**Review by RedXanadu — APPROVED** [review]
Change makes sense. New test looks good.

**Pull request merged**